File: src/main/java/com/es2/vadebicicleta/externo/email/controller/EmailControlller.java
package com.es2.vadebicicleta.externo.email.controller;

import com.es2.vadebicicleta.externo.email.dto.RequisicaoEmailConverter;
import com.es2.vadebicicleta.externo.email.dto.RequisicaoEmailInDto;
import com.es2.vadebicicleta.externo.email.model.RequisicaoEmail;
import com.es2.vadebicicleta.externo.email.service.EmailService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping
public class EmailControlller {
    @Autowired
    private final RequisicaoEmailConverter requisicaoEmailConverter;
    @Autowired
    private EmailService emailService;


    @Autowired
    public EmailControlller(RequisicaoEmailConverter requisicaoEmailConverter) {
        this.requisicaoEmailConverter = requisicaoEmailConverter;
    }

    @PostMapping("/enviarEmail")
    public ResponseEntity<Object> enviarEmail(@RequestBody RequisicaoEmailInDto dto){
        RequisicaoEmail emailEnviado = emailService.enviarEmail(requisicaoEmailConverter.inDtoToOject(dto));

        return ResponseEntity.ok(emailEnviado);
    }
}

File: src/main/kotlin/com/es2/vadebicicleta/externo/email/controller/EmailControlller.kt
package com.es2.vadebicicleta.externo.email.controller

import com.es2.vadebicicleta.externo.commons.dto.DtoConverter
import com.es2.vadebicicleta.externo.email.controller.dto.RequisicaoEmailInDto
import com.es2.vadebicicleta.externo.email.controller.dto.RequisicaoEmailOutDto
import com.es2.vadebicicleta.externo.dominio.RequisicaoEmail
import com.es2.vadebicicleta.externo.email.service.EmailService
import jakarta.validation.Valid
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping
class EmailControlller (
    private val requisicaoEmailConverter: DtoConverter<RequisicaoEmail, RequisicaoEmailInDto, RequisicaoEmailOutDto>,
    private val emailService: EmailService
) {
    @PostMapping("/enviarEmail")
    fun enviarEmail(@Valid @RequestBody dto: RequisicaoEmailInDto): ResponseEntity<RequisicaoEmailOutDto> {
        val emailEnviado = emailService.enviarEmail(requisicaoEmailConverter.toObject(dto))
        val outDto = requisicaoEmailConverter.toDto(emailEnviado)
        return ResponseEntity.ok(outDto)
    }
}
